stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2878,6 +2878,12 @@ LogicalResult MapOp::inferReturnTypeComponents(
     MLIRContext*, Optional<Location> location, ValueShapeRange operands,
     DictionaryAttr attributes, RegionRange regions,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+  // This function can be called directly from auto-gen builder(), and does not
+  // guarantee its arg `odsState.regions` to be non-empty. Need check it here to
+  // avoid `adaptor.getComputation()` crash.
+  if (regions.empty())
+    return emitOptionalError(location, "expect non-empty computation region");
```

Comment:
I think that we need a different kind of safeguard - the one that checks that the individual region is not `empty()`. The resultless builder does add regions, e.g.:

```C++
void MapOp::build(::mlir::OpBuilder &odsBuilder, ::mlir::OperationState &odsState, ::mlir::ValueRange inputs, ::mlir::DenseIntElementsAttr dimensions) {
  odsState.addOperands(inputs);
  odsState.addAttribute(getDimensionsAttrName(odsState.name), dimensions);
  (void)odsState.addRegion();

  ::llvm::SmallVector<::mlir::Type, 2> inferredReturnTypes;
  if (::mlir::succeeded(MapOp::inferReturnTypes(odsBuilder.getContext(),
                odsState.location, odsState.operands,
                odsState.attributes.getDictionary(odsState.getContext()),
                odsState.regions, inferredReturnTypes)))
    odsState.addTypes(inferredReturnTypes);
  else
    ::llvm::report_fatal_error("Failed to infer result type(s).");
}
```

What the resultless builder doesn't is populating these regions with operations, and that's something that we'll need to check for.

Reply:
Oh thanks, fixed.